src/04-modality-specific-files/05-task-events.md

```diff
@@ -93,29 +114,76 @@ Example:
-of example.
+Note that all other columns SHOULD also be described but are omitted for the
+sake of example.
+
+## Stimulus presentation details
+
+Specific fields related to stimulus presentation details SHOULD also be present
+in the accompanying JSON sidecar:
+
+| Field name           | Description                                                                                                                                                                                                                                                        |
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| StimulusPresentation | RECOMMENDED. [Object][] containing key value pairs related to the software used to present the stimuli during the experiment, specifically: `OperatingSystem`, `SoftwareName`, `SoftwareRRID`, `SoftwareVersion` and `Code`. See table below for more information. |
+
+The object supplied for `StimulusPresentation` SHOULD include the following key-value pairs:
+
+| Key name        | Description                                                                                                                                                                                                                   |
+|-----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| OperatingSystem | RECOMMENDED. Operating system used to run the stimuli presentation software (for formatting recommendations, see examples below this table).                                                                                  |
+| SoftwareName    | RECOMMENDED. Name of the software that was used to present the stimuli.                                                                                                                                                       |
+| SoftwareRRID    | RECOMMENDED. [Research Resource Identifier](https://scicrunch.org/resources) of the software that was used to present the stimuli. Examples: The RRID for Psychtoolbox is 'SCR_002881', and that of PsychoPy is 'SCR_006571'. |
+| SoftwareVersion | RECOMMENDED. Version of the software that was used to present the stimuli.                                                                                                                                                    |
+| Code            | RECOMMENDED. DOI of the archive (for example on Zenodo, figshare, etc.) for the code that was used to present the stimuli. Alternatively this could be a URL to code on GitHub, GitLab, etc.                                  |
```

Comment:
DOI or URL basically means URI. Maybe we can shorten this and give some guidance on how to ensure an unambiguous link?

```suggestion
| Code            | RECOMMENDED. URI of the code used to present the stimuli. Persistent identifiers such as DOIs are preferred. If multiple versions of code may be hosted at the same location, revision-specific URIs are recommended.     |
```

Reply:
I am in favor of this but maybe this "guidance" would be more appropriate for the whole spec, no?

Also a lot of people do not know the acronym URI: I know we are not planning to turn the specs into some sort of page turner / best seller... But it is important to keep that kind of consideration in mind.

Reply:
> I am in favor of this but maybe this "guidance" would be more appropriate for the whole spec, no?

We could turn URI into a link throughout the spec. And that link would lead users to a concise definition of what a URI is. ... The [wikipedia](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) page on that seems a bit long and convoluted for that purpose though.

Reply:
Or have our own section on this and use an internal link?

Reply:
Or maybe that's a job for the bids-starter kit ?

Reply:
> Or maybe that's a job for the bids-starter kit ?

If you are willing to write a short section on it, then that'd be a neat solution IMHO

Reply:
I guess the [glossary](https://github.com/bids-standard/bids-starter-kit/wiki/Glossary) in the wiki of BIDS starter kit is the most obvious place.

Although I suspect we will want to bring the wiki under proper version control in the future (right ?@dorahermes @franklin-feingold @emdupre) and might mean we will need to change that link in the future.

Will make an addition to the glossary for now anyway.

Would it make sense to open an issue for making URL, DOI and URI consistent across the specs? (as I believe this is beyond the scope of this PR).

Reply:
> Would it make sense to open an issue for making URL, DOI and URI consistent across the specs? (as I believe this is beyond the scope of this PR

+1 for out of the scope, and +1 to discuss it in a new issue and see whether or not it makes sense

Reply:
Updated bids-starter kit glossary: [link to U letter section](https://github.com/bids-standard/bids-starter-kit/wiki/Glossary#u)

This can be improved but I guess we can modify the content independently of this PR now.

Reply:
Mhh I would be all in favor of adding `[URI][uri]` and then putting `[uri]: https://github.com/bids-standard/bids-starter-kit/wiki/Glossary#u` right below `[object]: https://www.json.org/json-en.html` at the end of the file

BUT --> with the current setup of the glossary in BIDS starter kit, we cannot link to a dedicated keyword ... e.g., no link to URI, only to "U" and all that's listed there :thinking: ... it may be okay for now, but it doesn't seem ideal to me.

Do you have a solution for that @Remi-Gau? If not it'd be also fine to link to "U" and improve in the future.

Reply:
OK will move the link at the bottom. Makes sense.

I could use headings in the glossary to create links for each entry. The glossary is not huge yet, so still "easily" doable. That would be better, no?